src/retry-queue.js

```diff
@@ -0,0 +1,84 @@
+import { RequestQueueScaffold } from './base-request-queue'
+import { encodePostData, xhr } from './send-request'
+
+export class RetryQueue extends RequestQueueScaffold {
+    constructor(captureMetrics) {
+        super()
+        this.captureMetrics = captureMetrics
+        this.isPolling = false
+        this.queue = []
+        this.areWeOnline = true
+
+        if ('onLine' in window.navigator) {
+            this.areWeOnline = window.navigator.onLine
+            window.addEventListener('online', () => {
+                this._handleWeAreNowOnline()
+            })
+            window.addEventListener('offline', () => {
+                this.areWeOnline = false
+            })
+        }
+    }
+
+    enqueue(requestData) {
+        const retriesPerformedSoFar = requestData.retriesPerformedSoFar || 0
+        if (retriesPerformedSoFar >= 10) {
+            return
```

Comment:
From a previous conversation: https://github.com/PostHog/posthog-js/pull/226#discussion_r636812249

This basically means all retries have failed, maybe a `console.error` here?